### PR TITLE
Separate InlineToolbar UI and business logic

### PR DESCRIPTION
**Pull request by gohabereg**

### CODE REVIEW

File: packages/core/src/ui/InlineToolbar/index.ts [hunk 1]
```diff
@@ -1,204 +1,182 @@
...
+   */
+  #updateToolsList(tools: Map<InlineToolName, InlineTool>): void {
+    this.#nodes.buttons.innerHTML = '';
+
+    Array.from(tools.entries()).forEach(([name, tool]) => {
+      const button = make('button');
+
+      button.textContent = name;
+
+      if (Object.hasOwnProperty.call(tool.constructor.prototype, 'renderActions')) {
```
Comment (neSpecc) [inline packages/core/src/ui/InlineToolbar/index.ts:150]: there is ".hasActions()" in a facade

Comment (gohabereg) [reply packages/core/src/ui/InlineToolbar/index.ts:150]: Facade is not available here

File: packages/core/src/tools/internal/inline-tools/link/index.ts [hunk 2]
```diff
@@ -87,11 +87,6 @@ export default class LinkInlineTool implements InlineTool {
 
     linkInput.addEventListener('keydown', (event: KeyboardEvent) => {
       if (event.key === 'Enter') {
-        /**
-         * Remove link input, when data formed and trigger callback
-         */
-        linkInput.remove();
-
```
Comment (e11sy) [inline packages/core/src/tools/internal/inline-tools/link/index.ts:94]: ?why this change is needed?

File: packages/core/src/ui/InlineToolbar/index.ts (hunk 1 above)
Comment (neSpecc) [reply packages/core/src/ui/InlineToolbar/index.ts:150]: you can remove this method from the facade then

**Review by neSpecc — APPROVED** [review]

File: packages/core/src/api/SelectionAPI.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+import { SelectionManager } from '../components/SelectionManager.js';
+import { createInlineToolName } from '@editorjs/model';
+import { InlineToolFormatData } from '@editorjs/sdk';
+
+/**
+ * Selection API class
+ * - provides methods to work with selection
+ */
+@Service()
```
[the remark this answers is not in the document]
Comment (gohabereg) [reply packages/core/src/api/SelectionAPI.ts:12]: Anybody who needs it

[the remark this answers is not in the document]
Comment (gohabereg) [reply packages/core/src/api/SelectionAPI.ts:12]: Anybody who needs it, that's public API

File: packages/core/src/tools/internal/inline-tools/link/index.ts (hunk 2 above)
Comment (gohabereg) [reply packages/core/src/tools/internal/inline-tools/link/index.ts:94]: Just to keep link input there, doesn't matter atm as UX will be updated anyway

**Review by e11sy — APPROVED** [review]
